@ndlib/gatsby-theme-marble/src/components/Common/form.js

```diff
@@ -0,0 +1,137 @@
+import React, { Component } from 'react';
+import InputText from '../Common/inputText'
+import InputTextRadio from '../Common/inputRadio'
+import Recaptcha from 'react-google-invisible-recaptcha'
+
+import style from './style.module.css'
+import InputBox from './inputBox';
+
+class Form extends Component {
+  state = { 
+    data: {},
+    errors: {},
+    value: '',
+   }
+
+   validate = () => {
+    const errors = {}
+
+    const { data } = this.state
+
+    if (data.email.trim() === '')
+      errors.email = 'Email is required'
+    if (!data.email.includes('@')) 
+      return 'Email requires an @'
+    if (data.feedback.trim() === '')
+      errors.feedback = 'Please enter your comment or question'
+    if (data.category.trim() === '')
+      errors.category = 'A selection is required'
+
+    return Object.keys(errors).length === 0 ? null : errors
+  }
+
+  validateProperty = ({ name, value }) => {
+    if (name === 'email') {
+      if (value.trim() === '') return 'Email is required'
+      if (!value.includes('@')) return 'Email requires an @'
+    }
+  }
+
+  handleSubmit = e => {
+    e.preventDefault()
+
+    const errors = this.validate()
+    this.setState({ errors: errors || {} })
+    if (errors) return
+
+    if ( '' == this.state.value ) {
+      alert( 'Validation failed! Input cannot be empty.' );
+      this.recaptcha.reset();
+    } else {
+      this.recaptcha.execute();
+    }
+
+    this.doSubmit()
+  }
+
+  handleChange = ({ currentTarget: input }) => {
+    const errors = { ...this.state.errors }
+    const errorMessage = this.validateProperty(input)
+    if (errorMessage) errors[input.name] = errorMessage
+    else delete errors[input.name]
+
+    const data = {...this.state.data}
+    data[(input.name)] = (input.id === input.name 
+      ? input.value : input.id)
+
+    this.setState({ data, errors })
+  }
+
+  renderButton(label) {
+    return <button onClick={this.handleSubmit} disabled={this.validate()} className={style.button}>
+      {label}
+      </button>
+  }
+
+  renderInputText(name, label, autoFocus) {
```

Comment:
I'm not a huge fan of this pattern for rendering components. It unnecessarily mixes more javascript syntax in your presentation, and you don't get the advantage of calling it with named props that a functional component has. It's also a bit redundant, as these functions aren't really doing anything besides passing parameters down. That makes for more maintenance work.

I think the calling page should be able to render an `<InputText>` component directly rather than call a basic function which happens to return a component. I get that you want to wrap up the validation in a central place, so I would have the Form export a function that the individual components can import. Or pass down the validate function to children either with props or context. (See here: https://stackoverflow.com/questions/32370994/how-to-pass-props-to-this-props-children)